server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -65,7 +65,8 @@
   /**
    * If true {@link KillUnusedSegments} sends kill tasks for unused segments in all data sources.
    */
-  private final boolean killUnusedSegmentsInAllDataSources;
+  @Nullable
+  private final Boolean killUnusedSegmentsInAllDataSources;
```

Comment:
nit: can we compute its default per the given `killDataSourceWhitelist` instead of making it nullable? It could simplify the code a bit.